z3/src/context.rs

```diff
@@ -105,69 +108,71 @@ impl Context {
-        Ast::fresh_const(self, prefix, &self.bool_sort())
+    pub fn fresh_bool_const<'ctx>(&'ctx self, prefix: &str) -> ast::Bool<'ctx> {
+        ast::Bool::fresh_const(self, prefix)
     }
 
-    pub fn named_int_const(&self, s: &str) -> Ast {
-        Ast::new_const(&self.str_sym(s), &self.int_sort())
+    pub fn named_int_const(&self, s: &str) -> ast::Int {
+        ast::Int::new_const(&self.str_sym(s))
     }
 
-    pub fn numbered_int_const(&self, i: u32) -> Ast {
-        Ast::new_const(&self.int_sym(i), &self.int_sort())
+    pub fn numbered_int_const(&self, i: u32) -> ast::Int {
+        ast::Int::new_const(&self.int_sym(i))
     }
 
-    pub fn fresh_int_const<'ctx>(&'ctx self, prefix: &str) -> Ast<'ctx> {
-        Ast::fresh_const(self, prefix, &self.int_sort())
+    pub fn fresh_int_const<'ctx>(&'ctx self, prefix: &str) -> ast::Int<'ctx> {
+        ast::Int::fresh_const(self, prefix)
     }
 
-    pub fn named_real_const(&self, s: &str) -> Ast {
-        Ast::new_const(&self.str_sym(s), &self.real_sort())
+    pub fn named_real_const(&self, s: &str) -> ast::Real {
+        ast::Real::new_const(&self.str_sym(s))
     }
 
-    pub fn numbered_real_const(&self, i: u32) -> Ast {
-        Ast::new_const(&self.int_sym(i), &self.real_sort())
+    pub fn numbered_real_const(&self, i: u32) -> ast::Real {
+        ast::Real::new_const(&self.int_sym(i))
     }
 
-    pub fn fresh_real_const<'ctx>(&'ctx self, prefix: &str) -> Ast<'ctx> {
-        Ast::fresh_const(self, prefix, &self.real_sort())
+    pub fn fresh_real_const<'ctx>(&'ctx self, prefix: &str) -> ast::Real<'ctx> {
+        ast::Real::fresh_const(self, prefix)
     }
 
-    pub fn named_bitvector_const(&self, s: &str, sz: u32) -> Ast {
-        Ast::new_const(&self.str_sym(s), &self.bitvector_sort(sz))
+    pub fn named_bitvector_const(&self, s: &str, sz: u32) -> ast::BV {
+        ast::BV::new_const(&self.str_sym(s), sz)
     }
 
-    pub fn numbered_bitvector_const(&self, i: u32, sz: u32) -> Ast {
-        Ast::new_const(&self.int_sym(i), &self.bitvector_sort(sz))
+    pub fn numbered_bitvector_const(&self, i: u32, sz: u32) -> ast::BV {
+        ast::BV::new_const(&self.int_sym(i), sz)
     }
 
-    pub fn fresh_bitvector_const<'ctx>(&'ctx self, prefix: &str, sz: u32) -> Ast<'ctx> {
-        Ast::fresh_const(self, prefix, &self.bitvector_sort(sz))
+    pub fn fresh_bitvector_const<'ctx>(&'ctx self, prefix: &str, sz: u32) -> ast::BV<'ctx> {
+        ast::BV::fresh_const(self, prefix, sz)
     }
 
-    pub fn from_bool(&self, b: bool) -> Ast {
-        Ast::from_bool(self, b)
+    pub fn from_bool(&self, b: bool) -> ast::Bool {
+        ast::Bool::from_bool(self, b)
     }
 
-    pub fn from_u64(&self, u: u64) -> Ast {
-        Ast::from_u64(self, u)
+    // Deprecate from_u64() and from_i64() in favor of
+    // ast::Int::from_u64()/from_i64() and ast::BV::from_u64()/from_i64()
+    pub fn from_u64(&self, u: u64) -> ast::Int {
+        ast::Int::from_u64(self, u)
```

Comment:
this is another deprecation question

Reply:
Deprecate / remove.

Reply:
These turn out to be used a bunch even in the existing tests, so I'll just deprecate.